### DISCUSSION TITLE
Why the drop-down with the current task in the correction message?

### DISCUSSION THREAD

**Discussion by matthias-ronge** [kitodo/kitodo-production#4403] [Q&A] [answered]
When creating a correction message, a drop-down is offered in which the current task is displayed and selected. There are no alternatives to choose from here either. In which constellation is this selection significant?

![a drop-down with the current task](https://user-images.githubusercontent.com/3040657/105054782-a8d16980-5a72-11eb-9e20-6c720da4f8db.png)

**Comment by solth** [comment, marked as answer]
The current implementation of "Correction comments" requires a "correction task" (to which the workflow jumps back) and a "current task" (to which the workflow returns once the correction has been completed). 

Kitodo.Production 2 only supported strictly sequential lists of tasks in a process. Therefore, it was unambiguous which task was the "current" task at any given moment. But since Kitodo.Production 3 supports workflows with _parallel_ tasks, it is now possible to have multiple tasks open or in work at the same time. 

When opening a process from the **Current Task Details** page, the current task is self-explanatory - which is then preset as the "Current task" in the pulldown menu of the popup, btw - no matter how many parallel tasks might exist.

When opening a process with multiple "open" or "inwork" tasks in the metadata editor from the **Process List** however, it is **not** clear which of those parallel tasks is the "Current task", which is why we need the pulldown menu in the popup dialog.

Maybe this is an edge case, but it needs to be supported nonetheless (we already had such cases!)

Perhaps it would be a good solution to always deactivate the pulldown menu for "Current task" if there is only one option anyway (with an explanatory tooltip)?

**Comment by matthias-ronge** [comment]
So, _current task_ is meant as “after the correction task, return to…”?

If it is this, it makes a lot of sense and then it should offer all possibilities _after_ the task selected as correction task, _up to and including_ the current task. If this is how it is meant, the box should be relabeled (i.e. "after correction, return to") and I would also suggest to put it below the “correction task” select then. (In 2.x, I think the person who performed the correction task had possibility to select where to continue when finishing the correction task. But I am not sure. However, he or she _needs to know_, so it feels to me that the 3.x solution is better.)
